packages/mask/src/plugins/Savings/protocols/AAVEProtocol.ts

```diff
@@ -239,13 +239,14 @@ export class AAVEProtocol implements SavingsProtocol {
             poolAddress || ZERO_ADDRESS,
             AaveLendingPoolABI as AbiItem[],
         )
-        return new Promise<string>((resolve) =>
+        return new Promise<string>((resolve, reject) =>
             contract?.methods
```

Comment:
Please take `packages/mask/src/plugins/Gitcoin/hooks/useDonateCallback.ts` as an example. Refactor this to use `connection` for emitting transactions.